packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/DynamicFromObject.kt

```diff
@@ -0,0 +1,74 @@
+package com.facebook.react.bridge
+
+import com.facebook.common.logging.FLog
+import com.facebook.react.common.ReactConstants
+
+public class DynamicFromObject(private var mObject: Any?) : Dynamic {
+
+  override fun recycle() {
+    // Noop - nothing to recycle since there is no pooling
+  }
+
+  override val isNull: Boolean
+    get() = mObject == null
+
+
+  override fun asBoolean(): Boolean {
+    if (mObject !is Boolean) {
+      throw ClassCastException("Dynamic value from Object is not a boolean")
+    }
+    return mObject as Boolean
+  }
```

Comment:
Can you try to avoid the unnecessary casting? you could turn the logic around to do something like this:

```suggestion
  override fun asBoolean(): Boolean {
    if (mObject is Boolean) {
      return mObject
    }
    throw ClassCastException("Dynamic value from Object is not a boolean")
  }
```

Same for the rest

Reply:
Done